### ISSUE TITLE
Shortening file names in Icon style view 

### ISSUE THREAD

**Issue by 4r1y4n** [IgnorantGuru/spacefm#631]
Is it possible to shorten long file names in Icons style?  I mean some files have very long file names and spacefm shows the entire name that makes lots of wasted space between icons.  
Is it possible to make spacefm to for example show just n characters/lines of file name under the icon?

**Comment by IgnorantGuru** [comment]
No, this is not supported - I'll leave it open as a request.  Closest would be using List view and shortening the Name column.

If you want to try hacking it in, you might be able to add these lines:
~~~~
 g_object_set( G_OBJECT( renderer ),
        "ellipsize", PANGO_ELLIPSIZE_MIDDLE, NULL );
~~~~

to the else block in src/ptk/ptk-file-browser.c immediately after [this g_object setting](https://github.com/IgnorantGuru/spacefm/blob/a8955d06927a6ccce186316dd7a7d3754cf5c8ca/src/ptk/ptk-file-browser.c#L4577) ending at line 4584 (so your addition should come just before the '}' line of the else block).  Or you can use PANGO_ELLIPSIZE_END instead of MIDDLE.

That might run into some formatting and sizing issues though.

**Comment by clemencyworld1** [comment]
That would be a nice feature
